prereise/gather/flexibilitydata/doe/doe_data.py

```diff
@@ -1,85 +1,108 @@
+
+    :param string root: the root directory containing raw de-compressed DOE flexibility data
+    :return: None
+    """
 
     all_files = os.listdir(root)
     for i in all_files:
-        fp =  root+'/'+i
+        fp =  os.path.join(root, i)
         if i[0] == '.' or (i[-3:]=='csv' and os.path.getsize(fp) < 0.01):
             os.remove(fp)
 
     all_folders = os.listdir(root)
     for i in all_folders:
 
-        if not i[-3:] == 'csv':
-            cleanup_directory(root+i)
+        if os.path.isdir(os.path.join(root, i)):
+            cleanup_directory(os.path.join(root, i))
 
     return
 
-'''
-    download_doe()
 
-Download demand flexibility filters from OEDI, extract and cleanup    
-'''
-def download_doe():
+
+def download_doe(download_path='data'):
+    """Download demand flexibility filters from OEDI, extract and cleanup   
+
+    :param string download_path: the directory where the original DOE data will be stored
+    :return: None
+    """
 
     # create data directory 
-    if not os.path.exists('data'):
-        os.mkdir('data')
+    os.makedirs(download_path, exist_ok=True)
 
     # download zip data
     oedi_filter_link = "https://data.openei.org/files/180/2006weatherentireusdrfilters.tar.zip"
-    urllib.request.urlretrieve(oedi_filter_link, 'data/filter.zip')
+    urllib.request.urlretrieve(oedi_filter_link, download_path+'/filter.zip')
 
     # extract
-    with zipfile.ZipFile('data/filter.zip', 'r') as fh:
-        fh.extractall('data')
+    with zipfile.ZipFile(download_path+'/filter.zip', 'r') as fh:
+        fh.extractall(download_path)
 
     # delete and further extract
-    os.remove('data/filter.zip')
+    os.remove(download_path+'/filter.zip')
 
-    for i in os.listdir('data/'):
-        fh = tarfile.open('data/'+i)
-        fh.extractall('data')
+    for i in os.listdir(download_path):
+        fh = tarfile.open(os.path.join(download_path, i))
+        fh.extractall(download_path)
 
         fh.close()
-        os.remove('data/'+i)
+        os.remove(os.path.join(download_path, i))
 
     # cleanup
-    cleanup_directory('data/')
-
+    cleanup_directory(download_path)
 
-'''
-    aggregate_doe(root, out)
 
-Aggregate sector flexibilties by summing up the percentage flexibility from all sectors
-and store to output files
-'''
-def aggregate_doe(root, out):
-    if not os.path.exists(out):
-        os.mkdir(out)
+def aggregate_doe(root, out_path):
+    """Aggregate sector flexibilties by summing up the percentage flexibility from all sectors
+        and store to output csv file
+        
+    :param string root: the root directory containing raw de-compressed DOE flexibility data
+    :param string out_path: the output file where the aggregated data will be stored
+    :return: None
+    """
 
     all_folders = os.listdir(root)
+    
+
+    # initialize output container
+    eia_flex = pd.DataFrame(np.zeros((8808, len(all_folders))), columns=all_folders)
```

Comment:
number of hours for most files in the DOE data set. I think this code does not need to be generalized because it is only supposed to deal with this specific dataset though

Reply:
I wonder if it's the number of hours required in local time to represent a full year in UTC time. 24 * 367 = 8808.

Reply:
What is the time frame that they are looking at? There are only 8784 hours in a leap year (366 days)

Reply:
For whatever reason this includes the last day of 2005 and the 1st day of 2007......

Reply:
That makes sense to me. One day from previous year + one day from next year will give a full year in wherever local time conversion.